hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -169,13 +170,14 @@ private void openServerSockets() {
             serverSocket.setReceiveBufferSize(receiveBufferSize);
             serverSocket.setReuseAddress(true);
             port = bind(serverSocket, port, limit);
-            serverSocket.accept(socket -> {
+            serverSocket.accept(acceptRequest -> {
+                AsyncSocket socket = reactor.openAsyncSocket(acceptRequest);
                 socket.setReadHandler(readHandlerSuppliers.get(reactor).get());
                 socket.setSendBufferSize(sendBufferSize);
                 socket.setReceiveBufferSize(receiveBufferSize);
                 socket.setTcpNoDelay(tcpNoDelay);
                 socket.setKeepAlive(true);
```

Comment:
Can't we send these into the openAsyncSocket() call so that these can also be final? For example imagine sending `receiveBufferSize` into `reactor.openAsyncSocket()` so that returned socket have all final fields.

Reply:
I had similar thoughts. What can be done is to pass a config object; similar to what we do in hz classic networking. Netty does something similar.